hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/validators/block/StateChangesValidator.java

```diff
@@ -245,16 +251,28 @@ public StateChangesValidator(
                 new ConfigProviderImpl().getConfiguration(),
                 networkInfo,
                 new NoOpMetrics());
-
         logger.info("Registered all Service and migrated state definitions to version {}", servicesVersion);
     }
 
     @Override
     public void validateBlocks(@NonNull final List<Block> blocks) {
         logger.info("Beginning validation of expected root hash {}", expectedRootHash);
+        var previousBlockHash = INITIAL_BLOCK_HASH;
+
         for (final var block : blocks) {
+            StateChanges lastStateChanges = null;
+            // get the state hash before applying the state changes from current block
+            //            CRYPTO.digestTreeSync(state);
+            final var startOfBlockStateHash = Bytes.wrap(new byte[48]); // state.getHash();
```

Comment:
Not sure I understand the comment here...meant to be this?
```suggestion
            // (FUTURE) use `state.getHash()` instead of an empty byte array
            final var startOfBlockStateHash = Bytes.wrap(new byte[48]); 
```